File: BlazeDemo/pom/pageObjects/CartPage.py
class CartPage:

    def __init__(self, page):
        self.page = page

    def go_product_detail(self):
        self.page.locator("//*[@id='tbodyid']/div[1]/div/div/h4/a").click()

    def add_product(self):
        self.page.locator("//*[@id='tbodyid']/div[2]/div/a").click()

    def open_cart(self):
        self.page.locator("#cartur").click()

    def get_product_title(self):
        return self.page.locator("//*[@id='tbodyid']/tr/td[2]")

    def get_product_price(self):
        return self.page.locator("//*[@id='tbodyid']/tr/td[3]")

    def click_delete(self):
        self.page.locator("//*[@id='tbodyid']/tr/td[4]/a").click()

File: PlayWrightAutomationJS/pageobjects/CartPage.js

class CartPage {

    constructor(page)
    {
        this.page = page;
        this.prodDetail = page.locator("//*[@id='tbodyid']/div[1]/div/div/h4/a");
        this.addProduct = page.locator("//*[@id='tbodyid']/div[2]/div/a");
        this.openCart = page.locator("#cartur");
        this.productTitle = page.locator("//*[@id='tbodyid']/tr/td[2]");
        this.productPrice = page.locator("//*[@id='tbodyid']/tr/td[3]");
        this.deleteBtn = page.locator("//*[@id='tbodyid']/tr/td[4]/a");
    }

    async clickProductDetails(){
        await this.prodDetail.click();
    }

    async addProductBtn(){
        await this.addProduct.click();
    }

    async openCartPage(){
        await this.openCart.click();
    }

    async deleteProduct(){
        await this.deleteBtn.click();
    }

}

module.exports = {CartPage}